add limitations (#892)
returntocorp/semgrep-docs: docs/supported-languages.md

To install and run Semgrep Pro Engine, see [Semgrep Pro Engine overview](/deepsemgrep/deepsemgrep-introduction/).

> ### Known Limitations

Reviewer: ```suggestion
### Known limitations
```

> <b>CommonJS</b>

Reviewer: ```suggestion
#### CommonJS
```

> Currently Semgrep Pro does not handle all cases of CommmonJS imports, if you create a function and assign it to an export later Semgrep Pro does not capture this today, read [this page](https://semgrep.dev/docs/deepsemgrep/deepsemgrep-examples/#es6-and-commonjs) for more information.

Reviewer: ```suggestion
Currently Semgrep Pro does not handle all cases of CommonJS imports. See [Known limitations](/deepsemgrep/deepsemgrep-examples/#known-limitations) in Semgrep Pro Engine documentation.
```

Reviewer: No need to repeat the same information - we can either modularize it or link to it.

> <b>Regressions in Semgrep Pro</b>

Reviewer: ```suggestion
#### Regressions in Semgrep Pro
```

> For interfile analysis, Semgrep Pro Engine resolves names differently than Semgrep and this may cause rules with `interfile: true` to produce different results than Semgrep OSS Engine. Some instances could be regarded as regressions, if you encounter them please file a bug report.

Reviewer: @LewisArdern where do we recommend users submit Semgrep Pro Engine bug reports?

Reviewer: @LewisArdern @IagoAbal to specify: In which platform do we recommend users to submit tickets? Should they submit issues to regular Sermgrep OSS repository even for Semgrep Pro Engine issues?

> 

Reviewer: ```suggestion
```

> 
> 

Reviewer: ```suggestion
```

To install and run Semgrep Pro Engine, see [Semgrep Pro Engine overview](/deepsemgrep/deepsemgrep-introduction/).

### Known limitations of Semgrep Pro Engine

#### CommonJS

Currently Semgrep Pro Engine does not handle specific cases of CommmonJS where you define a function and assign it to an export later, Semgrep Pro Engine does not track the code below:

```js
function get_user() {
    return get_user_input("example")
  }

module.exports = get_user
```

#### Regressions in Semgrep Pro

> For interfile analysis, Semgrep Pro Engine resolves names differently than Semgrep and this may cause rules with `interfile: true` to produce different results than Semgrep OSS Engine. Some instances could be regarded as regressions, if you encounter them please file a bug report.

Reviewer: ```suggestion
For interfile analysis, Semgrep Pro Engine resolves names differently than Semgrep OSS and this may cause rules with `interfile: true` to produce different results than Semgrep OSS Engine. Some instances could be regarded as regressions, if you encounter them please file a bug report.
```

Reviewer: https://github.com/returntocorp/semgrep-docs/pull/892#discussion_r1149067891 @LewisArdern

> For interfile analysis, Semgrep Pro Engine resolves names differently than Semgrep OSS. Consequently, rules with `interfile: true` may produce different results than Semgrep OSS Engine. Some instances could be regarded as regressions, if you encounter them please file a bug report.

Reviewer: ```suggestion
For interfile analysis, Semgrep Pro Engine resolves names differently than Semgrep OSS. Consequently, rules with `interfile: true` may produce different results than Semgrep OSS Engine. Some instances could be regarded as regressions, if you encounter them please file a bug report. When you need to report a bug in Semgrep Pro Engine, go through [support@r2c.dev](mailto:support@r2c.dev). You can also contact us through [Semgrep Community Slack group](https://go.semgrep.dev/slack).
```
